internal/metrics/metrics.go

```diff
@@ -209,3 +214,95 @@ func getStatusMetrics(subsystem string, labels []string) map[string]prometheus.C
 		),
 	}
 }
+
+// ObjCounter creates and registers a new CounterVec metric with the given name and help
+// text. The returned CounterVec embeds the CounterVec from the prometheus package and adds a method
+// to increment the counter for a given object. The labels of the metric are determined from the
+// name and the namespace of the given object.
+func ObjCounter(name, help string) (c ObjCounterVec) {
+	labels := []string{"name", "namespace"}
```

Comment:
I'm worried about tracking objects individidually could turn this into a "high-cardinality" metrics on big scale systems.

Reply:
I don't see a problem with those two labels. It's basically "one" metric per GitRepo. Even if a customer had a thousand GitRepos, I don't think that this would cause any problems for Prometheus. I said "one" because the combination of both name and namespace is the identifier. The label values aren't arbitrary and do not change for the whole lifetime of the GitRepo. The metric itself denotes the type, e.g. `gitrepo`. I wouldn't consider that high cardinality. Also, this allows for insights as to which GitRepo might be causing problems, or at least how long it takes to download and process them, as opposed to having that information without the ability to connect that with any particular GitRepo.

Where do your concern come from?

Reply:
Whether its high-cardinality or not is actually very subjective and depend on a case-by-case basis. I'm just pointing out this could create an unexpected situation for users with a bigger amount of GitRepos.
I understand there is value being able to get per-object insights, but we could argue that Prometheus may not be the best choice for showing those stats. Part of this information may belong to the logs, while would offer an aggregated view of it?
At the very least, I think we should provide a way to configure this behavior (not sure what the default value should be).

Reply:
FTR, after talking about this offline, it turns out that these changes are consistent with the existent metrics, but we agreed it's not ideal. @p-se will create a separate issue to look into this from a broader perspective.